packages/blade/src/components/Card/_decisions/decisions.md

```diff
@@ -0,0 +1,512 @@
+    <CardHeader.Counter />
+  </Card.HeaderLeading>
+  <Card.HeaderTrailing>
+    <Badge />
+  </Card.HeaderTrailing>
+</Card.Header>
+```
+
+**API 6:** Constrained Prop Based: 
+
+**Pros:** 
+- Gives us total control of how we want to structure the Card
+No footguns as such
+
+**Cons:** 
+- Prop based API seems a bit too restrictive
+- Hard to extend later on for future usecases, imagine if we want to add Avatar we need to add `prefixAvatar` prop for it
+- Passing props to counter needs another prop `counterIntent`
+
+```jsx
+<Card>
+  <Card.Header
+    title=""
+    subtitle=""
+    prefixIcon={CloseIcon}
+    prefixAvatar={}
+    counter={12}
+    counterIntent=""
+    trailingVisual={<Badge />}
+  />
+</Card>
+```
+
+**API 7: Final API**
+
+After discussing about the various pros & cons, footguns & advantages of the APIs we proposed a hybrid approach: 
+
+
+**Pros:**
+- Gives us control over which props to expose in components like `Card.Header.Icon`
+- Breaking change resistent, The API is open for extension, if we want we can expose `Card.Header.Avatar` without introducing a breaking change
+- Retains all the advantages of the previous hybrid approach.
+- No runtime hacks to ensure the order of components unlike in the compound component approach
+- We can constrain the titlePrefix, titleSuffix to only accept `Card.Header.{component}`
+
+**Cons:**
+- Treeshaking will be a minor issue, if we have 10 `Card.Header.{component}`s then all 10 will be included in the bundle even if users don't use them
+
+
+```jsx
+<Card>
+  <Card.Header 
+    titlePrefix={<Card.Header.Icon />}
+    titleSuffix={<Card.Header.Counter />}
+    title="Title"
+    subTitle="Subtitle" 
+  />
+</Card>
+```
+
+> **Internal Meeting: Dec 5th 2022**
+> 
+> **Agenda:** Discussing TreeShaking issue with the Final API
+> 
+> Participants: Anurag, Kamlesh, Chaitanya, Abinash, Saurabh, Saurav
+
+In this meeting we discussed about the 1 issue with the final API which was Treeshaking. 
+
+
+After discussing about the criticality of the treeshaking we decided to go with the approach where we treeshaking is possible.
+
+- Reason being in future there can be more components like `Card.Header.Avatar` which will further increase the bundle
+  - If someone just uses the Card even if they are not using any of the other components like Button, IconButton, Badge their app will bloat with the components
+- We also discussed few approaches on how we can validate the components and throw a runtime error if an unallowed component is passed to the Card. 
+  - We cannot use displayName of React components since it will be stripped of on production
+  - We also cannot do `comp.type === CardHeaderCounter` because that way we anyway have to import the component, resulting in bloated bundle
+
+
+
+**Action Items:** 
+
+- Try adding a static method to the Card components to determine if it’s an allowed component or not. We can’t use displayName since it will be stripped off.
```

Comment:
This needs to be updated I think with the latest approach.

Reply:
Yeah i've updated the API, but kept this here for archiving the discussions we had while developing the API.